src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/IssueCreateDialog.kt

```diff
@@ -72,24 +72,32 @@ class IssueCreateDialog(
      *
      */
     fun showAllFields(): IssueCreateDialog {
-        val configureFieldsDialogId = "inline-dialog-field_picker_popup"
-        val dialogLocator = By.id(configureFieldsDialogId)
+        val configureFieldsDialogClass = "qf-picker"
+        val dialogLocator = By.className(configureFieldsDialogClass)
+
         try {
             openConfigureFieldsDialog(dialogLocator)
         } catch (e: TimeoutException) {
             //we probably sometimes click too fast, but no idea what we should wait for
             openConfigureFieldsDialog(dialogLocator)
         }
-        val locator = By.xpath("//div[@id='$configureFieldsDialogId']//dd[1]//a")
-        if (driver.isElementPresent(locator)) {
-            driver.wait(elementToBeClickable(locator)).click()
-            driver.wait(invisibilityOfElementLocated(dialogLocator))
-            driver.wait(visibilityOfElementLocated(dialogLocator))
+
+        val allFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
+        if (driver.isElementPresent(allFieldsLinkLocator)) {
+            driver.wait(elementToBeClickable(allFieldsLinkLocator)).click()
+            val newAllFieldsLinkLocator = getVisibleAllFieldsLinkLocator(configureFieldsDialogClass)
+            driver.wait(not(elementToBeClickable(newAllFieldsLinkLocator)))
         }
+
         dismissConfigureFieldsDialog()
         return this
     }
 
+    private fun getVisibleAllFieldsLinkLocator(dialogClass: String) : By {
+        val sectionIndex = if (driver.findElement(By.cssSelector(".$dialogClass dl:nth-of-type(1)")).isDisplayed) 1 else 2
+        return By.xpath("//div[@class='$dialogClass']//dl[$sectionIndex]//dd[1]//a")
```

Comment:
Ah, to dream about semantic HTML 😉 
Perhaps the frontend team could make the markup more robot-friendly? I asked 🤞

Reply:
Yup. :) Another option would be to have shared page objects, but that sound too crazy.

Reply:
When we started JPT, we wanted to reuse the `atlassian-jira-pageobjects` (or something), but it came with a few problems:
* 300+ MB of dependencies to put into the shadow JAR
* Maven POM magic to autoinclude a Chrome/Firefox executable based on your current OS
    * they were out of date fast - e.g. Chome 30
    * if you run JPT on Windows, you don't want the `googlechrome.exe` on the VU node which runs Ubuntu
* despite honest effort, we couldn't figure out how to actually run this code - too many magical annotations and reflection
* very restrictive patterns - [page objects are a flexible pattern](https://github.com/SeleniumHQ/selenium/wiki/PageObjects)
* not sure if they are publicly distributed